lib/collection/src/save_on_disk.rs

```diff
@@ -57,6 +57,27 @@ impl<T: Serialize + for<'de> Deserialize<'de> + Clone> SaveOnDisk<T> {
         })
     }
 
+    /// Initialize data with `init`, even if data already exists.
+    ///
+    /// Will immediately save the new `init` state on disk, only if the file already exists.
+    pub fn init(path: impl Into<PathBuf>, init: impl FnOnce() -> T) -> Result<Self, Error> {
+        let path: PathBuf = path.into();
+        let exists = path.exists();
+        let data = Self {
+            change_notification: Condvar::new(),
+            notification_lock: Default::default(),
+            data: RwLock::new(init()),
+            path,
+        };
+
+        // Overwrite existing data if it exists
+        if exists {
+            data.save()?;
+        }
+
+        Ok(data)
+    }
```

Comment:
```suggestion
    /// Initialize new data, even if it already exists on disk at the given path.
    ///
    /// If data already exists on disk, it will be immideately overwritten.
    pub fn new(path: impl Into<PathBuf>, data: T) -> Result<Self, Error> {
        let data = Self {
            change_notification: Condvar::new(),
            notification_lock: Default::default(),
            data: RwLock::new(data),
            path: path.into(),
        };

        if data.path.exists() {
            data.save()?;
        }

        Ok(data)
    }
```